test/framework/arm-test-utils.js

```diff
@@ -0,0 +1,527 @@
+  this.numberOfRandomTestIdGenerated = 0;
+  //stub necessary methods if in playback mode
+  this._stubMethods();
+}
+
+_.extend(ArmTestUtils.prototype, {
+  _stubMethods: function () {
+    if (this.isPlayback) {
+      sinon.stub(msRestAzure.UserTokenCredentials.prototype, 'signRequest', function (webResource, callback) {
+        return callback(null);
+      });
+
+      sinon.stub(msRestAzure.ApplicationTokenCredentials.prototype, 'signRequest', function (webResource, callback) {
+        return callback(null);
+      });
+
+      sinon.stub(this, 'createResourcegroup', function (groupName, location, callback) {
+        return callback(null);
+      });
+
+      sinon.stub(this, 'deleteResourcegroup', function (groupName, callback) {
+        return callback(null);
+      });
+    }
+  },
+  
+  _setupResourceManagementClient: function () {
+   this.resourceManagement = new ResourceManagementClient(this.createUserCredentials(), this.subscriptionId);
+   if (this.isPlayback) {
+    this.resourceManagement.longRunningOperationRetryTimeoutInSeconds = 0;
+   }
+  },
+
+  createResourcegroup: function (groupName, location, callback) {
+    console.log('Creating Resource Group: ' + groupName);
+    //if (!this.isPlayback) {
+      if (!this.resourceManagement) { this._setupResourceManagementClient(); }
+      return this.resourceManagement.resourceGroups.createOrUpdate(groupName, {'location': location}, callback);
+    //}
+    //return callback(null);
+  },
+
+  deleteResourcegroup: function (groupName, callback) {
+    console.log('Deleting Resource Group: ' + groupName);
+    //if (!this.isPlayback) {
+      if (!this.resourceManagement) { this._setupResourceManagementClient(); }
+      return this.resourceManagement.resourceGroups.deleteMethod(groupName, callback);
+    //}
+    //return callback(null);
+  },
+
+  /**
+   * Creates a new UserTokenCredentials object.
+   * 
+   * @returns {ms-rest-azure.UserTokenCredentials} The user token credentials object
+   */
+  createUserCredentials: function () {
+    return new msRestAzure.UserTokenCredentials(this.clientId, this.domain, this.username, this.password, this.clientRedirectUri, { 'tokenCache': this.tokenCache });
+  },
+  
+  /**
+   * Creates a new ApplicationTokenCredentials object.
+   * 
+   * @returns {ms-rest-azure.ApplicationTokenCredentials} The application token credentials object
+   */
+  createApplicationCredentials: function () {
+    return new msRestAzure.ApplicationTokenCredentials(this.clientId, this.domain, this.secret, { 'tokenCache': this.tokenCache });
+  },
+  
+  /**
+   * Provides the recordings directory for the test suite
+   *
+   * @returns {string} The test recordings directory
+   */
+  getRecordingsDirectory: function () {
+    return this.recordingsDirectory;
+  },
+  
+  /**
+   * Provides the recordings directory for the test suite
+   *
+   * @returns {string} The test recordings directory
+   */
+  getTockenCacheFilePath: function () {
```

Comment:
I suggest not to create the help method unless it is really needed. This can be folded into the constructor as well.